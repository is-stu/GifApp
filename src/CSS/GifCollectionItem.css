.card {
  border-radius: 20px;
  border: 1px solid black;
  margin: 10px 10px;
  max-height: 300px;
  max-width: 200px;
  overflow: hidden;
  text-align: center;
  width: fit-content;
  background-color: white;
}

.img {
  width: 200px;
  height: 200px;
}

.card p {
  padding: 10px 10px;
  text-transform: capitalize;
}
