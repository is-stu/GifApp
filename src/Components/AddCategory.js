import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, categories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setCategories([inputValue, ...categories]);
    setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type='text' value={inputValue} onChange={handleChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
