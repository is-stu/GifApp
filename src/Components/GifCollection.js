import React, { useEffect, useState } from 'react';
import { GifCollectionItem } from './GifCollectionItem';
import '../CSS/GifCollection.css';

export const GifCollection = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=LTFMP1Xw9xadj7IIELLcsj2zD65TAOhB`;
    const res = await fetch(url);
    const { data } = await res.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    // console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className='cardsContainer'>
        {images.map((img) => (
          <GifCollectionItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
