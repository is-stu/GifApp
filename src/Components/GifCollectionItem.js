import React from 'react';
import '../CSS/GifCollectionItem.css'

export const GifCollectionItem = ({ id, title, url }) => {
  //   console.log({ id, title, url });

  return (
    <div className='card'>
      <img src={url} alt={title} className='img' />
      <p> {title} </p>
    </div>
  );
};
