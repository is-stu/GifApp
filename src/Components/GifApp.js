import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifCollection } from './GifCollection';

export function GifApp() {
  const [categories, setCategories] = useState(['Rick & Morty']);

  //   const handleAdd = (e) => {
  //     e.preventDefault();
  //     setCategories([...categories, 'otro']);
  //   };

  return (
    <div>
      <h2>GifApp</h2>
      <AddCategory setCategories={setCategories} categories={categories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifCollection category={category} key={category} />
        ))}
      </ol>
    </div>
  );
}
