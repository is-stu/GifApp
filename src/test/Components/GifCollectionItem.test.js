import { shallow } from 'enzyme';
import React from 'react';
import { GifCollectionItem } from '../../Components/GifCollectionItem';

describe('Testing GifCollectionItem', () => {
  test('Match with Snapshot ', () => {
    const wrapper = shallow(<GifCollectionItem />);

    expect(wrapper).toMatchSnapshot();
  });
});
