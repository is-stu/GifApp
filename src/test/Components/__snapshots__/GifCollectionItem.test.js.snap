// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifCollectionItem Match with Snapshot  1`] = `
<div
  className="card"
>
  <img
    className="img"
  />
  <p>
     
     
  </p>
</div>
`;
